ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PauseResumeIntegrationTest.java

```diff
@@ -0,0 +1,279 @@
+        + "QUERIES;").get(0)).getQueries().get(0).getId().toString();
+    RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "PAUSE " + queryId + ";");
+    assertThat(getPausedCount(), equalTo(1));
+
+    // Produce more records
+    TEST_HARNESS.produceRows(PAGE_VIEW_TOPIC, PAGE_VIEWS_PROVIDER2, KAFKA, JSON,
+        System::currentTimeMillis);
+    
+    assertThatEventually(supplier, equalTo(9));
+
+    // Then:
+    RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "RESUME " + queryId + ";");
+
+    // 5 more records have been produced
+    assertThatEventually(supplier, equalTo(14));
+
+    List <KsqlEntity> showQueries3 = RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "SHOW "
+        + "QUERIES;");
+    assertThat(((Queries) showQueries3.get(0)).getQueries().get(0).getStatusCount().getStatuses()
+        .get(KsqlQueryStatus.RUNNING), equalTo(1));
+  }
+
+  @Test
+  public void pausedQueriesShouldBePausedOnRestart()  {
+    // Given:
+    createQuery("1");
+    createQuery("2");
+
+    TEST_HARNESS.produceRows(PAGE_VIEW_TOPIC, PAGE_VIEWS_PROVIDER, KAFKA, JSON,
+        System::currentTimeMillis);
+
+    Supplier<Integer> supplier = getSupplier(1);
+    Supplier<Integer> supplier2 = getSupplier(2);
+
+    // 7 records are produced + a header & footer.
+    assertThatEventually(supplier, equalTo(9));
+    assertThatEventually(supplier2, equalTo(9));
+
+    // When:
+    String queryId = ((Queries) RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "SHOW "
+        + "QUERIES;").get(0)).getQueries().get(0).getId().toString();
+    RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "PAUSE "
+        + queryId + ";");
+    assertThatEventually(supplier, equalTo(9));
+
+    // Restart server
+    REST_APP.stop();
+    REST_APP.start();
+
+    // Verify PAUSED state -- eventually
+    assertThatEventually(this::getPausedCount, equalTo(1));
+
+    // Verify number of processed records
+    assertThatEventually(supplier, equalTo(9));
+
+    // Then:
+    String queryId2 = ((Queries) RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "SHOW "
+        + "QUERIES;").get(0)).getQueries().get(0).getId().toString();
+    RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "RESUME " + queryId2 + ";");
+    assertThatEventually(this::getPausedCount, equalTo(0));
+    assertThatEventually(this::getRunningCount, equalTo(1));
+
+    // Adding more after restart.
+    TEST_HARNESS.produceRows(PAGE_VIEW_TOPIC, PAGE_VIEWS_PROVIDER2, KAFKA, JSON,
+        System::currentTimeMillis);
+
+    assertThatEventually(supplier, equalTo(14));
+
+    List <KsqlEntity> showQueries3 = RestIntegrationTestUtil.makeKsqlRequest(REST_APP, "SHOW "
+        + "QUERIES;");
+    assertThat(((Queries) showQueries3.get(0)).getQueries().get(0).getStatusCount().getStatuses()
+        .get(KsqlQueryStatus.RUNNING), equalTo(1));
+  }
+
+  // Parameterize this entire class to run with multiple REST instances?
+  // @Test
```

Comment:
I don't think this test is necessary. It'd essentially be re-testing that the command topic distributes commands as intended, which is outside the scope of pause/resume.